cmd/keyloader.go

```diff
@@ -17,50 +17,50 @@ package cmd
 import (
 	"context"
 	"fmt"
+	"strings"
 
+	"github.com/spf13/pflag"
 	"github.com/testifysec/go-witness/cryptoutil"
-	"github.com/testifysec/go-witness/signer/file"
-	"github.com/testifysec/go-witness/signer/fulcio"
-	"github.com/testifysec/go-witness/signer/spiffe"
+	"github.com/testifysec/go-witness/signer"
 	"github.com/testifysec/witness/options"
 )
 
-func loadSigners(ctx context.Context, ko options.KeyOptions) ([]cryptoutil.Signer, []error) {
-	signers := []cryptoutil.Signer{}
-	errors := []error{}
+// signerProvidersFromFlags looks at all flags that were set by the user to determine which signer providers we should use
+func signerProvidersFromFlags(flags *pflag.FlagSet) map[string]struct{} {
+	signerProviders := make(map[string]struct{})
+	flags.Visit(func(flag *pflag.Flag) {
+		if !strings.HasPrefix(flag.Name, "signer-") {
+			return
+		}
 
-	//Load key from fulcio
-	if ko.FulcioURL != "" {
-		fulcioSigner, err := fulcio.Signer(ctx, ko.FulcioURL, ko.OIDCClientID, ko.OIDCIssuer, ko.Token)
-		if err != nil {
-			err := fmt.Errorf("failed to create signer from Fulcio: %w", err)
-			errors = append(errors, err)
-		} else {
-			signers = append(signers, fulcioSigner)
+		parts := strings.Split(flag.Name, "-")
+		if len(parts) < 2 {
+			return
 		}
-	}
 
-	//Load key from file
-	if ko.KeyPath != "" {
-		fileSigner, err := file.Signer(ctx, ko.KeyPath, ko.CertPath, ko.IntermediatePaths)
+		signerProviders[parts[1]] = struct{}{}
+	})
+
+	return signerProviders
+}
+
+// loadSigners loads all signers that appear in the signerProviders set and creates their respective signers, using any options provided in so
+func loadSigners(ctx context.Context, so options.SignerOptions, signerProviders map[string]struct{}) ([]cryptoutil.Signer, error) {
+	signers := make([]cryptoutil.Signer, 0)
+	for signerProvider := range signerProviders {
+		setters := so[signerProvider]
+		sp, err := signer.NewSignerProvider(signerProvider, setters...)
 		if err != nil {
-			err := fmt.Errorf("failed to create signer from file: %w", err)
-			errors = append(errors, err)
-		} else {
-			signers = append(signers, fileSigner)
+			return signers, fmt.Errorf("failed to create %v signer provider: %w", signerProvider, err)
```

Comment:
Is it worth continuing instead of returning here?  If not, should `signers` be `nil` instead?

Reply:
Yeah, it probably makes more sense to continue here and just log this